<a href="https://colab.research.google.com/github/lexxxegorov/HOMEWORKPY/blob/master/Lesson_8.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.

In [ ]:
import numpy as np
import scipy.stats as stats

In [ ]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

In [ ]:
cov = np.mean(zp*ks) - np.mean(zp) * np.mean(ks)
cov

9157.839999999997

In [ ]:
np.cov(zp,ks, ddof = 0)

array([[ 3494.64,  9157.84],
       [ 9157.84, 30468.89]])

In [37]:
np.std(zp, ddof = 0)


59.115480206118605

In [ ]:
np.std(ks, ddof = 0)

174.55340157098058

In [ ]:
9157.84 / (59.115480206118605 * 174.55340157098058)

0.8874900920739162

In [ ]:
np.corrcoef(zp,ks)

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [ ]:
iq = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])

In [ ]:
x_1 = np.mean(iq)
x_1

118.1

In [ ]:
np.std(iq, ddof=1)

10.54566788359614

In [ ]:
t1 = stats.t.ppf(0.975, 9)
t1

2.2621571627409915

In [ ]:
x_1 - t1 * (10.54566788359614 / np.sqrt(10))

110.55608365158724

In [ ]:
x_1 + t1 * (10.54566788359614 / np.sqrt(10))

125.64391634841274

**доверительный интервал для математического ожидания [110.57 ; 125.64]**

Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического
ожидания с надежностью 0.95.

In [ ]:
174.2 - 1.96 * np.sqrt(25/27)

172.31398912064722

In [ ]:
174.2 + 1.96 * np.sqrt(25/27)

176.08601087935276

**доверительный интервал для математического ожидания [172.31 ; 176.09]**